google-cloud-datastore/src/main/java/com/google/cloud/datastore/DatastoreBatchWriter.java

```diff
@@ -37,12 +37,24 @@
 
   /**
    * {@inheritDoc}
+   * If {@code entity} has a complete key and was already marked for deletion in this writer, the
+   * operation will be changed to {@link #put}.
+   *
+   * @throws DatastoreException if a given entity with the same complete key was already added to
+   *     this writer, if writer is not active or if id allocation for an entity with an incomplete
+   *     key failed
+   */
+  @Override
+  Entity add(FullEntity<?> entity);
+
+  /**
+   * {@inheritDoc}
    * For entities with complete keys that were marked for deletion in this writer the operation
    * will be changed to {@link #put}.
    *
    * @throws DatastoreException if a given entity with the same complete key was already added to
    *     this writer, if writer is not active or if id allocation for an entity with an incomplete
-   *     key failed.
+   *     key failed
```

Comment:
According to the [javadoc guide](http://www.oracle.com/technetwork/java/javase/documentation/index-137868.html#@exception) it's not needed (same holds for `@param` and `@return` descriptions).